src/ImageSharp/Common/Helpers/SimdUtils.Shuffle.cs

```diff
@@ -182,9 +183,9 @@ private static void VerifyShuffle3SpanInput<T>(ReadOnlySpan<T> source, Span<T> d
             where T : struct
         {
             DebugGuard.IsTrue(
-                source.Length == dest.Length,
+                source.Length <= dest.Length,
                 nameof(source),
-                "Input spans must be of same length!");
+                "Source should fit into dest!");
```

Comment:
Did we have a scenario where they weren't the same length? When I wrote this `IShuffle3` and `IShuffle4` were mirrored implementations designed for working against specific pixel formats. `VerifyShuffle4SpanInput` still checks for equality.

Reply:
The original version fails [`EncodeBaseline_WorksWithDiscontiguousBuffers<Bgr24>`](https://github.com/SixLabors/ImageSharp/blob/424d4f94357f6efa66b436e59be367fe6b0c4dc6/tests/ImageSharp.Tests/Formats/Jpg/JpegEncoderTests.cs#L209) on current master, we just did not notice it since CI doesn't run tests in Debug (maybe it should?).

`this.rgbSpan` is longer than `this.pixelSpan` here:
https://github.com/SixLabors/ImageSharp/blob/659daffad916a20a5b62399089c212033c7494e4/src/ImageSharp/Formats/Jpeg/Components/Encoder/YCbCrForwardConverter420%7BTPixel%7D.cs#L109

because of the padding added by @br3aker:
https://github.com/SixLabors/ImageSharp/blob/659daffad916a20a5b62399089c212033c7494e4/src/ImageSharp/Formats/Jpeg/Components/Encoder/YCbCrForwardConverter420%7BTPixel%7D.cs#L88

I believe we should follow `Span.CopyTo(source, dest)` semantics in this family of methods, which allows `dest` to be longer and processes `source.Length` items.

Reply:
>CI doesn't run tests in Debug (maybe it should?).

Oh please no! They take long enough!!

>I believe we should follow `Span.CopyTo(source, dest)` semantics in this family of methods, which allows dest to be longer and processes `source.Length` items.

I agree then and we should update `VerifyShuffle4SpanInput ` to match.

Reply:
We can introduce a new `Checked` build configuration, like many C/C++ projects do. It's basically Release build but with all the checks/guards enabled.